### PR TITLE
Or fix

### PR DESCRIPTION
**Pull request by sjorn3**
In the end the problem was that the branches weren't being updated properly after creating an if. Should be correct now. There was another slightly subtler issue on the julia-0.7 branch for Charlotte that didn't help finding this, but both things should be good now.

### CODE REVIEW

File: src/passes.jl [hunk 1]
```diff
@@ -50,12 +34,38 @@ function branches_to(b)
   return result
 end
 
+# After removing branch, none are point to the branch, but decrement pointing
+# above.
```
Comment (tshort) [inline src/passes.jl]: This is a confusing comment.

Comment (sjorn3) [reply src/passes.jl]: Probably made sense to me at the time... I'll give it another go.

**Comment by tshort** [comment]
On tests, I'm good with tests of WebAssembly going into Charlotte. 

Anyway, looks good to me.

**Comment by MikeInnes** [comment]
The tests don't have to be end-to-end – if there's a minimal WASM block that reveals this error, we can just test that running the pass on it gives expected output.

**Pull request merged**